spec/services/tip_options_service_property_spec.rb

```
@@ -0,0 +1,168 @@
+    end
+
+    property 'accepts arrays with large integers' do
+      forall do
+        array_of(integer(min: -1_000_000, max: 1_000_000))
+      end.check do |options|
+        expect(TipOptionsService.send(:are_tip_options_valid?, options)).to be true
+      end
+    end
+  end
+
+  describe 'is_default_tip_option_valid?', :property_based do
```

RuboCop (1.75.5)

[convention] Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

(Style/StringLiterals)